main.go

```diff
@@ -55,13 +56,14 @@ func main() {
 		Ephemeral: true,
 		Hostname: "bonk",
 	}
+	defer s.Close()
 
 	tsclient, err := s.LocalClient()
 	if err != nil {
 		log.Fatal(err)
 	}
+	defer tsclient.Logout(context.Background())
```

Comment:
This was the missing piece: we need to logout once the program exits.

I tested this by `pkill -INT bonk` (after commenting out all the Mosyl stuff), and then changing the hostname, and there was no new machine, just a changed hostname. But we should probably test this properly before calling it good. Unfortunately, I don't have the requisite fly or TS creds, so this falls to you, @grahamc.